clang/unittests/Format/TokenAnnotatorTest.cpp

```diff
@@ -2499,6 +2499,15 @@ TEST_F(TokenAnnotatorTest, BraceKind) {
   EXPECT_BRACE_KIND(Tokens[6], BK_Block);
 }
 
+TEST_F(TokenAnnotatorTest, StreamOperator) {
+  auto Tokens = annotate("\"foo\\n\" << aux << \"foo\\n\" << \"foo\";");
+  ASSERT_EQ(Tokens.size(), 9u) << Tokens;
+  EXPECT_FALSE(Tokens[1]->MustBreakBefore);
+  EXPECT_FALSE(Tokens[3]->MustBreakBefore);
+  // Only break between string literals if the former ends with \n.
+  EXPECT_TRUE(Tokens[5]->MustBreakBefore);
+}
+
```

Comment:
Can you delete this test or move it to FormatTest.cpp?

Reply:
i added this test as it's more "unit" compared to the final formatting, which might be affected by all sorts of other things (column limits, bin-packing etc.). whereas this one particularly checks that we provide the "relevant" signal only in the case we care about.

we can't really have it in the FormatTest, as this is the file that has utilities for asserting on annotations of tokens.

if you think that isn't useful, i am happy to drop it completely, but i believe this is valuable for the reason i mentioned above.

Reply:
I understand, but `TokenAnnotatorTest` is mainly for testing token types and has never been used for checking `MustBreakBefore` and other similar token attributes. @HazardyKnusperkeks @rymiel @mydeveloperday what do you think?

Reply:
It hasn't been done. But I think it's the right place. The formatting test just tests that it breaks at a position. This here intents to test that it the flag `MustBreakBefore` is set.

I'm in favor of this kind of test, but don't insist on it.

Reply:
Ok, let’s keep it then.